### ISSUE TITLE
Variable IOS_USE_PRECOMPILED_FIRESTORE_POD does not work on 14.2.1-cli

### ISSUE THREAD

**Issue by BlaineM-SeriouslyRAD** [dpa99c/cordova-plugin-firebasex#759]
# Bug report

**CHECKLIST**
- [x] I have read the [issue reporting guidelines](https://github.com/dpa99c/cordova-plugin-firebasex#reporting-issues)
- [x] I confirm this is a suspected bug or issue that will affect other users
- [x] I have reproduced the issue using the [example project](https://github.com/dpa99c/cordova-plugin-firebasex-test) or provided the necessary information to reproduce the issue.
- [x] I have read [the documentation](https://github.com/dpa99c/cordova-plugin-firebasex) thoroughly and it does not help solve my issue.
- [x] I have checked that no similar issues (open or closed) already exist.

**Current behavior:**

Using the 14.2.1-**cli** build, the pre-compiled Firestore pod is **always** used regardless of the variable `IOS_USE_PRECOMPILED_FIRESTORE_POD`.

The non-cli 14.2.1 version works fine.

Comparing `plugin.xml` between 14.2.1-cli and 14.2.1 reveals **a number of differences** including:

14.2.1-cli:
`<pod name="FirebaseFirestore" git="https://github.com/invertase/firestore-ios-sdk-frameworks.git" tag="9.1.0"/>`

14.2.1
`<pod name="Firebase/Firestore" spec="9.1.0"/>`

**Expected behavior:**
The variable `IOS_USE_PRECOMPILED_FIRESTORE_POD` should control whether the pre-compiled Firestore pod is used.

**Steps to reproduce:**
Add 14.2.1-cli into a project without specifying the variable (defaults to false).
Check the PodFile - it states:
`pod 'FirebaseFirestore', :tag => '9.1.0', :git => 'https://github.com/invertase/firestore-ios-sdk-frameworks.git'`

**Environment information**
- Cordova CLI version 
	- `11.0.0`
- Cordova platform version
	- `ios 6.2.0`
- Dev machine OS and version, e.g.
    - OSX
        - `Monterey 12.5.1`
- Node JS version
    - `16.17.0`

**Comment by TiBz0u** [comment]
Hi @dpa99c, it seems the merge on plugin.xml fails.
The following [line](https://github.com/dpa99c/cordova-plugin-firebasex/blob/8607adf2efff6d279a3cc7d151ce38ef11687d22/plugin.xml#L150) must be updated to point directly to the official pod version 9.1.0 .
Thanks !

**Comment by BlaineM-SeriouslyRAD** [comment]
@TiBz0u Due to this issue with the cli version, we're using the non-cli version with the official pod (i.e. not pre-compiled). Consequently, we haven't experienced an issue with instant crashing on devices.

**Comment by RobYed** [comment]
Can confirm it doesn't work. Tried it as a fix for the error of #735. I wrote a cordova hook, to overwrite the Podfile. Similar to [this one](https://github.com/dpa99c/cordova-plugin-firebasex/issues/735#issuecomment-1180019463). Now it works.

However, would be great to see it fixed :)

**Comment by rex-iotum** [comment]
> I wrote a cordova hook to overwrite the Podfile. Similar to https://github.com/dpa99c/cordova-plugin-firebasex/issues/735#issuecomment-1180019463. Now it works.

@RobYed can you share what you added in that `after_prepare` hook?

**Comment by RobYed** [comment]
@rex-iotum sure! It's very basic with a hard-coded version number, but might help:

```js
module.exports = (context) => {
  // Make sure ios platform is part of build
  if (!context.opts.platforms.includes('ios')) {
    return;
  }

  const iosPlatformRoot = path.join(context.opts.projectRoot, 'platforms/ios');
  const podfilePath = path.join(iosPlatformRoot, 'Podfile');

  if (!fs.existsSync(podfilePath)) {
    console.log(
        `'${podfilePath}' does not exist. FirebaseFirestore fix skipped.`
    );
    return;
  }

  const preCompiledVersionString =
      "pod 'FirebaseFirestore', :tag => '9.1.0', :git => 'https://github.com/invertase/firestore-ios-sdk-frameworks.git'";

  let podfileContent = fs.readFileSync(podfilePath, 'utf-8');

  if (podfileContent.includes(preCompiledVersionString)) {
    podfileContent = podfileContent.replace(
        preCompiledVersionString,
        "pod 'FirebaseFirestore', '9.1.0'"
    );

    fs.writeFileSync(podfilePath, podfileContent, 'utf-8');

    console.log('Installing non pre-compiled FirebaseFirestore...');

    return execa('pod', ['install', '--verbose'], {
      cwd: iosPlatformRoot,
    });
  } else {
    console.log('Could not set non pre-compiled FirebaseFirestore');
  }
};
```

**Comment by TiBz0u** [comment]
Hi @RobYed.
Since release 14.2.0, the pre-compiled version is not the default anymore. As explained upper, there is a merge issue.
If you want the same behavior on the cli, instead of hook, juste adapt the plugin.xml to the right pod (as [here](https://github.com/dpa99c/cordova-plugin-firebasex/blob/3e56f531e25eeec350c429455f4fe09757731ae7/plugin.xml#L158))...

In all case, you'll need a fork before the fixing release. Or use the current_cli branch with has already the PR you asked
https://github.com/dpa99c/cordova-plugin-firebasex/pull/757

**Comment by dpa99c** [comment]
Should be fixed in `cordova-plugin-firebasex@15.0.0-cli` which has just been released to npm
